src/data/contexts/ActivitiesContext.tsx

```diff
@@ -0,0 +1,52 @@
+import { IndexableType } from 'dexie';
+import { Activity, Category } from '../interfaces';
+import {
+	useState,
+	Dispatch,
+	SetStateAction,
+	createContext,
+	ReactNode,
+} from 'react';
+import { useLiveQuery } from 'dexie-react-hooks';
+import { db } from '../Database';
+
+export interface IActivitiesContext {
+	activities: Activity[];
+	categories: Category[];
+	selectedCategories: IndexableType[];
+	setSelectedCategories: Dispatch<SetStateAction<IndexableType[]>>;
+}
+
+interface ContextProviderProps {
+	children: ReactNode;
+}
+
+export const ActivitiesContext = createContext<IActivitiesContext>({
+	activities: [],
+	categories: [],
+	selectedCategories: [],
+	setSelectedCategories: () => {},
+});
+
+export default function ContextProvider({ children }: ContextProviderProps) {
+	const activities =
+		useLiveQuery<Activity[]>(() => db.activities.toArray()) ?? [];
+	const categories =
+		useLiveQuery<Category[]>(() => db.categories.toArray()) ?? [];
```

Comment:
Pretty sure you don't need `?? []` here as the query already returns an empty array by default

Reply:
Refer to [this comment](https://github.com/Polarts/feel-tracker/issues/13#issuecomment-1527651116) as well